### PR TITLE
Remove legacy ruby implementation

### PR DESCRIPTION
**Pull request by anttijk**
#### 18e217e0c1998003d85e7596be9e58bb25867328
<pre>
Remove legacy ruby implementation
<a href="https://bugs.webkit.org/show_bug.cgi?id=270741">https://bugs.webkit.org/show_bug.cgi?id=270741</a>
<a href="https://rdar.apple.com/124325661">rdar://124325661</a>

Reviewed by Alan Baradlay.

Remove dead code.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::computeAccessibilityIsIgnored const):
(WebCore::AccessibilityRenderObject::determineAccessibilityRole):
* Source/WebCore/editing/Editing.cpp:
(WebCore::isBlock):
* Source/WebCore/editing/TextIterator.cpp:
(WebCore::shouldEmitNewlinesBeforeAndAfterNode):
* Source/WebCore/html/RubyElement.cpp:
(WebCore::RubyElement::createElementRenderer): Deleted.
* Source/WebCore/html/RubyElement.h:
* Source/WebCore/html/RubyTextElement.cpp:
(WebCore::RubyTextElement::createElementRenderer):
* Source/WebCore/html/track/WebVTTElement.cpp:
* Source/WebCore/layout/integration/LayoutIntegrationCoverage.cpp:
(WebCore::LayoutIntegration::canUseForChild):
(WebCore::LayoutIntegration::canUseForLineLayoutWithReason):
* Source/WebCore/layout/integration/LayoutIntegrationCoverage.h:
* Source/WebCore/platform/ios/SelectionGeometry.cpp:
(WebCore::SelectionGeometry::SelectionGeometry):
(WebCore::operator&lt;&lt;):
* Source/WebCore/platform/ios/SelectionGeometry.h:
(WebCore::SelectionGeometry::isInFixedPosition const):
(WebCore::SelectionGeometry::isRubyText const): Deleted.
* Source/WebCore/rendering/LegacyInlineFlowBox.cpp:
(WebCore::placeChildInlineBoxesInBlockDirection):
(WebCore::LegacyInlineFlowBox::computeOverAnnotationAdjustment const):
(WebCore::LegacyInlineFlowBox::computeUnderAnnotationAdjustment const):
* Source/WebCore/rendering/LegacyInlineTextBox.cpp:
* Source/WebCore/rendering/LegacyLineLayout.cpp:
(WebCore::LegacyLineLayout::computeExpansionForJustifiedText):
(WebCore::expansionBehaviorForInlineTextBox):
(WebCore::LegacyLineLayout::computeInlineDirectionPositionsForSegment):
(WebCore::LegacyLineLayout::layoutLineBoxes):
(WebCore::LegacyLineLayout::setMarginsForRubyRun): Deleted.
(WebCore::LegacyLineLayout::updateRubyForJustifiedText): Deleted.
* Source/WebCore/rendering/LegacyLineLayout.h:
* Source/WebCore/rendering/LegacyRootInlineBox.cpp:
(WebCore::LegacyRootInlineBox::selectionTop const):
(WebCore::LegacyRootInlineBox::selectionBottom const):
* Source/WebCore/rendering/RenderBlock.cpp:
(WebCore::RenderBlock::selectionGaps):
(WebCore::RenderBlock::blockSelectionGaps):
(WebCore::RenderBlock::baselinePosition const):
(WebCore::RenderBlock::firstLineBaseline const):
(WebCore::RenderBlock::lastLineBaseline const):
(WebCore::RenderBlock::inlineBlockBaseline const):
* Source/WebCore/rendering/RenderBlock.h:
* Source/WebCore/rendering/RenderBlockFlow.cpp:
(WebCore::RenderBlockFlow::willCreateColumns const):
(WebCore::RenderBlockFlow::firstLineBaseline const):
(WebCore::RenderBlockFlow::lastLineBaseline const):
(WebCore::RenderBlockFlow::inlineBlockBaseline const):
(WebCore::RenderBlockFlow::inlineSelectionGaps):
* Source/WebCore/rendering/RenderBlockFlow.h:
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::computeLogicalWidthInFragment const):
* Source/WebCore/rendering/RenderElementInlines.h:
(WebCore::RenderElement::shouldApplyLayoutOrPaintContainment const):
(WebCore::RenderElement::shouldApplySizeOrStyleContainment const):
* Source/WebCore/rendering/RenderImage.cpp:
(WebCore::RenderImage::collectSelectionGeometries):
* Source/WebCore/rendering/RenderLineBreak.cpp:
(WebCore::RenderLineBreak::collectSelectionGeometries):
* Source/WebCore/rendering/RenderObject.cpp:
(WebCore::RenderObject::collectSelectionGeometriesInternal):
* Source/WebCore/rendering/RenderObject.h:
(WebCore::RenderObject::isRenderRubyAsInline const): Deleted.
(WebCore::RenderObject::isRenderRubyAsBlock const): Deleted.
(WebCore::RenderObject::isRenderRubyBase const): Deleted.
(WebCore::RenderObject::isRenderRubyRun const): Deleted.
(WebCore::RenderObject::isRenderRubyText const): Deleted.
* Source/WebCore/rendering/RenderRuby.cpp: Removed.
* Source/WebCore/rendering/RenderRuby.h: Removed.
* Source/WebCore/rendering/RenderRubyBase.cpp: Removed.
* Source/WebCore/rendering/RenderRubyBase.h: Removed.
* Source/WebCore/rendering/RenderRubyRun.cpp: Removed.
* Source/WebCore/rendering/RenderRubyRun.h: Removed.
* Source/WebCore/rendering/RenderRubyText.cpp: Removed.
* Source/WebCore/rendering/RenderRubyText.h: Removed.
* Source/WebCore/rendering/RenderText.cpp:
(WebCore::RenderText::collectSelectionGeometries):
(WebCore::RenderText::emphasisMarkExistsAndIsAbove):
* Source/WebCore/rendering/RenderTreeAsText.cpp:
(WebCore::RenderTreeAsText::writeRenderObject):
* Source/WebCore/rendering/TextDecorationPainter.cpp:
(WebCore::collectStylesForRenderer):
* Source/WebCore/rendering/line/BreakingContext.h:
(WebCore::BreakingContext::handleReplaced):
(WebCore::BreakingContext::canBreakAtThisPosition):
(WebCore::BreakingContext::commitAndUpdateLineBreakIfNeeded):
* Source/WebCore/rendering/line/LineWidth.cpp:
(WebCore::LineWidth::updateLineDimension):
(WebCore::LineWidth::computeAvailableWidthFromLeftAndRight):
(WebCore::LineWidth::applyOverhang): Deleted.
* Source/WebCore/rendering/line/LineWidth.h:
* Source/WebCore/rendering/updating/RenderTreeBuilder.cpp:
(WebCore::RenderTreeBuilder::attachInternal):
(WebCore::RenderTreeBuilder::detach):
(WebCore::RenderTreeBuilder::destroyAndCleanUpAnonymousWrappers):
* Source/WebCore/rendering/updating/RenderTreeBuilderBlock.cpp:
(WebCore::canDropAnonymousBlock):
(WebCore::RenderTreeBuilder::Block::removeLeftoverAnonymousBlock):
* Source/WebCore/rendering/updating/RenderTreeBuilderFirstLetter.cpp:
(WebCore::supportsFirstLetter):
* Source/WebCore/rendering/updating/RenderTreeBuilderList.cpp:
(WebCore::getParentOfFirstLineBox):
* Source/WebCore/rendering/updating/RenderTreeBuilderRuby.cpp:
(WebCore::findRubyRunParent): Deleted.
(WebCore::isRubyChildForNormalRemoval): Deleted.
(WebCore::RenderTreeBuilder::Ruby::moveInlineChildren): Deleted.
(WebCore::RenderTreeBuilder::Ruby::moveBlockChildren): Deleted.
(WebCore::RenderTreeBuilder::Ruby::moveChildren): Deleted.
(WebCore::RenderTreeBuilder::Ruby::moveChildrenInternal): Deleted.
(WebCore::RenderTreeBuilder::Ruby::attach): Deleted.
(WebCore::RenderTreeBuilder::Ruby::findOrCreateParentForChild): Deleted.
(WebCore::RenderTreeBuilder::Ruby::rubyBaseSafe): Deleted.
(WebCore::RenderTreeBuilder::Ruby::detach): Deleted.
* Source/WebCore/rendering/updating/RenderTreeBuilderRuby.h:

Canonical link: <a href="https://commits.webkit.org/275879@main">https://commits.webkit.org/275879@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/b2245233df96590cc3975d231ec2f3a7220f42c0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/43152 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/22176 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/45552 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/45781 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/39275 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/45456 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/25925 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/19599 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/35676 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/43725 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/19229 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/37178 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/16674 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/16815 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/38244 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/1206 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/39353 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/38561 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/47323 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/18066 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/14856 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/42468 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/19603 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/41126 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9601 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/19781 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/19235 "Built successfully") | | |

### CODE REVIEW

File: Source/WebCore/css/CSSCrossfadeValue.cpp [hunk 1]
```diff
@@ -68,4 +68,15 @@ RefPtr<StyleImage> CSSCrossfadeValue::createStyleImage(Style::BuilderState& stat
...
+{
+    if (func(m_fromValueOrNone.get()) == IterationStatus::Done)
+        return IterationStatus::Done;
+    if (func(m_toValueOrNone.get()) == IterationStatus::Done)
+        return IterationStatus::Done;
+    if (func(m_percentageValue.get()) == IterationStatus::Done)
+        return IterationStatus::Done;
+    return IterationStatus::Continue;
+}
+
```
Comment (alanbaradlay) [inline Source/WebCore/css/CSSCrossfadeValue.cpp]: ?

File: Source/WebCore/css/CSSCrossfadeValue.h [hunk 2]
```diff
@@ -51,16 +51,7 @@ class CSSCrossfadeValue final : public CSSValue {
...
-    {
-        if (func(m_fromValueOrNone.get()) == IterationStatus::Done)
-            return IterationStatus::Done;
-        if (func(m_toValueOrNone.get()) == IterationStatus::Done)
-            return IterationStatus::Done;
-        if (func(m_percentageValue.get()) == IterationStatus::Done)
-            return IterationStatus::Done;
-        return IterationStatus::Continue;
-    }
+    IterationStatus customVisitChildren(const Function<IterationStatus(CSSValue&)>&) const;
```
Comment (alanbaradlay) [inline Source/WebCore/css/CSSCrossfadeValue.h]: ?

**Review by alanbaradlay — APPROVED** [review]

File: Source/WebCore/css/CSSCrossfadeValue.cpp (hunk 1 above)
Comment (anttijk) [reply Source/WebCore/css/CSSCrossfadeValue.cpp]: compile fix due to unified build, just moving a function to .cpp

File: Source/WebCore/css/CSSCrossfadeValue.h (hunk 2 above)
Comment (anttijk) [reply Source/WebCore/css/CSSCrossfadeValue.h]: see above

**Pull request merged**